docs/data/tree-view/tree-item-customization/useTreeItemModelHook.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import { RichTreeView } from '@mui/x-tree-view/RichTreeView';
+import { TreeItem, TreeItemProps } from '@mui/x-tree-view/TreeItem';
+import { TreeViewBaseItem } from '@mui/x-tree-view/models';
+import { useTreeItemModel } from '@mui/x-tree-view/hooks';
+
+type TreeItemWithLabel = {
+  id: string;
+  label: string;
+  isHighlighted?: boolean;
+};
+
+export const MUI_X_PRODUCTS: TreeViewBaseItem<TreeItemWithLabel>[] = [
```

Comment:
We could probably move this to a separate file for a shorter main demo 🤔 Or reuse the items from `./products`